src/codegen/compile.ml

```diff
@@ -10312,6 +10312,47 @@ and compile_exp_as env ae sr_out e =
   let sr_in, code = compile_exp_with_hint env ae (Some sr_out) e in
   code ^^ StackRep.adjust env sr_in sr_out
 
+and single_case e (cs : Ir.case list) =
+  match cs, e.note.Note.typ with
+  | [{it={pat={it=TagP (l, _);_}; _}; _}], Type.(Variant [{lab; _}]) -> l = lab
+  | _ -> false
+
+and simplify_cases e (cs : Ir.case list) =
+  match cs, e.note.Note.typ with
+  (* for a 2-cased variant type, the second second comparison can be omitted when the first pattern
+     (with irrefutable subpattern) didn't match *)
+  | [{it={pat={it=TagP (_, ip);_}; _}; _} as c1; {it={pat={it=TagP (_, pat');_} as pat2; exp}; _} as c2], Type.(Variant [_; _])
+       when Ir_utils.is_irrefutable ip ->
+     [c1; {c2 with it = {exp; pat = {pat2 with it=TagP ("", pat')}}}]
+  | _ -> cs
+
+
+
+and _ascending_remainders env e =
```

Comment:
Please disregard this function, I'll remove (will hopefully come back in a more modular fashion in a follow-up PR).